modules/core/src/main/scala/com/romanidze/kaesquella/core/models/ExecutionError.scala

```diff
@@ -0,0 +1,45 @@
+package com.romanidze.kaesquella.core.models
+
+import tethys._
+import tethys.jackson._
+import tethys.derivation.semiauto._
```

Comment:
Where do you use semiauto? I see that you implement readers and writers manually.